submodule change: introduce google bench

yokofly opened the pull request:
_update v1: option title`ENABLE_BENCHMARK` -> `ENABLE_BENCHMARKS`_
_update v2: post clickhouse community pr for tracking_

disable by default, use option `-DENABLE_BENCHMARKS=ON/OFF` 

fix #147 
upstream pr https://github.com/ClickHouse/ClickHouse/pull/43779

yokofly committed:
submodule change: introduce google bench
(disable by default)

yokofly commented on src/Common/examples/CMakeLists.txt:
```diff
@@ -85,3 +85,8 @@ endif()
+
+if (ENABLE_BENCHMARK)
+    proton_add_executable (benchmark_string_copy benchmark_string_copy.cpp)
+    target_link_libraries (benchmark_string_copy PRIVATE dbms)
```
todo:
1. move to src/Common/benchmarks folder
2. link necessary lib instead whole dbms archive

yokofly committed:
sync with upstream submodule name and example code

yokofly committed:
Merge remote-tracking branch 'origin/develop' into feature/issue-147-introduce-google-bench

yokofly committed:
Merge branch 'develop' into feature/issue-147-introduce-google-bench

yokofly committed:
update google-benchmark into git modules

yokofly committed:
fix incorrect file path in cmake

yokofly committed:
Merge branch 'develop' into feature/issue-147-introduce-google-bench

qijun-niu-timeplus commented on build.sh:
```diff
@@ -44,6 +44,7 @@ cmake .. \
     -DENABLE_CLANG_TIDY=OFF \
     -DENABLE_TESTS=ON \
     -DENABLE_EXAMPLES=ON \
+    -DENABLE_BENCHMARKS=ON \
```
I don't think we should recommand others to compiler benchmarks by default.
Could you set it `OFF` here?

yokofly replied:
this is `build.sh` /

yokofly replied:
above DENABLE_EXAMPLES and DENABLE_TESTS all turn to ON, so I do the same

yokofly replied:
my point is:
1. this file is not obviously recommended to build/use right?
2. the example has been set on, it's straightforward i also build the benchmark right? any special reasons to set off?
even when we build the program, those are all executable binary, we still need manual `./benchmark_xxx` to call it.

yokofly replied:
discussed internally:
> The ENABLE_EXAMPLES && ENABLE_TESTS options are useful for development,
but ENABLE_BENCHMARKS is optional.
Having it on by default enlarges the link time for **dbms**,
so disabling it is recommended.

yokofly committed:
Disable `ENABLE_BENCHMARKS` in build.sh

The ENABLE_EXAMPLES && ENABLE_TESTS options are useful for development,
but ENABLE_BENCHMARKS is optional.
Having it on by default enlarges the link time for dbms,
so disabling it is recommended.
